base/abstractset.jl

```diff
@@ -476,6 +476,27 @@ function isdisjoint(a, b)
     _isdisjoint(a, b)
 end
 
+function isdisjoint(a::AbstractRange{T}, b::AbstractRange{T}) where T
+    (isempty(a) || isempty(b)) && return true
+    fa, la = extrema(a)
+    fb, lb = extrema(b)
+    if (la < fb) | (lb < fa)
+        return true
+    else
+        return _overlapping_range_isdisjoint(a, b)
+    end
+end
+
+_overlapping_range_isdisjoint(a::AbstractRange{T}, b::AbstractRange{T}) where T = invoke(isdisjoint, Tuple{Any,Any}, a, b)
+
+function _overlapping_range_isdisjoint(a::AbstractRange{T}, b::AbstractRange{T}) where T<:Integer
+    if abs(step(a)) == abs(step(b))
+        return mod(minimum(a) - minimum(b), step(a)) != 0
```

Comment:
This could overflow. Here are three examples where it does and produces the wrong answer:
```julia
isdisjoint(0x1:0x3:0x4, 0x4:0x3:0x4)
isdisjoint(0x3:0x3:0x6, 0x4:0x3:0x4)
isdisjoint(typemax(Int):3:typemax(Int), typemin(Int):3:typemax(Int))
```
Ranges are so hard, sorry.